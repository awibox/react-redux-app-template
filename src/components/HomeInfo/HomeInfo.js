import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ImmutablePropTypes from 'react-immutable-proptypes';
// Components
import { Button, Card } from 'tigerspack';
// Styles
import styles from './HomeInfo.scss';

// Using "Stateless Functional Components"
export default class HomeInfo extends Component {
  static propTypes = {
    home: ImmutablePropTypes.contains({
      name: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      html_url: PropTypes.string.isRequired,
    }),
  };

  buttonClick = (url) => window.open(url);

  render() {
    const { home } = this.props;
    return (
      <div className={styles.homeInfo}>
        <Card theme={'light'}>
          <div className={styles.homeLogo}>
            <div className={styles.image}>
              <i className="fa fa-gears"></i>
            </div>
            <h1 className={styles.name}>{home.get('name')}</h1>
            <div className={styles.text}>{home.get('description')}</div>
            <div className={styles.button}>
              <Button icon={<i className="fa fa-github"/>} onClick={() => this.buttonClick(home.get('html_url'))}>
                Download with Github
              </Button>
            </div>
          </div>
        </Card>
      </div>
    );
  }
}
