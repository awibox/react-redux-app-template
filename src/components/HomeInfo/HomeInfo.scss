@import '../../styles/variables';


.homeInfo {
  text-align: center;
}

.homeLogo {
  display: flex;
  flex-flow: column;
  .image {
    font-size: 128px;
    padding: $defaultIndent;
    color: $logoColor;
  }
  .name {
    line-height: 56px;
    font-size: 56px;
    font-weight: 100;
    padding-top: 10px;
  }
  .text {
    padding: 20px;
    font-size: 20px;
    color: #333;
    font-weight: 100;
    line-height: 30px;
    margin-bottom: 30px;
  }
}

.version {
  font-size: 18px;
  line-height: 22px;
  text-align: left;
  margin-left: 5px;
}

.button {
  padding-bottom: 20px;
}
