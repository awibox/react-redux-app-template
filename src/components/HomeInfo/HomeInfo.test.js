import React from 'react';
import { Map } from 'immutable';
import { shallow } from 'enzyme';
import HomeInfo from './HomeInfo';

const mockData = Map({
  name: 'Finder',
  description: 'Search engine',
  html_url: 'http://localhost:8888/url',
});

describe('HomeInfo', () => {
  it('should render correctly', () => {
    const component = shallow(<HomeInfo home={mockData} />);
    expect(component).toMatchSnapshot();
  });
});
