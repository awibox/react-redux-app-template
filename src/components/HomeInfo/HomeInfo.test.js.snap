// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeInfo should render correctly 1`] = `
<div
  className="homeInfo"
>
  <Card
    indent={15}
    padding={15}
    theme="light"
  >
    <div
      className="homeLogo"
    >
      <div
        className="image"
      >
        <i
          className="fa fa-gears"
        />
      </div>
      <h1
        className="name"
      >
        Finder
      </h1>
      <div
        className="text"
      >
        Search engine
      </div>
      <div
        className="button"
      >
        <Button
          icon={
            <i
              className="fa fa-github"
            />
          }
          indent={0}
          onClick={[Function]}
        >
          Download with Github
        </Button>
      </div>
    </div>
  </Card>
</div>
`;
